### PR TITLE
feature/6004-theo-InAppRecruitmentAnalytics

### PR DESCRIPTION
**Pull request by theodur**
## Description of Change

Adds analytics for in-app recruitment.

## Testing

- [ ] Tested on iOS 
- [ ] Tested on Android 

## Reviewer Validations

-  [ ] `vama_givefb_close`: user clicks "Close" button in "Give feedback" large panel (**or swipes the panel away?**)
-  [ ] `vama_givefb_launch`: User clicks "Get started" button in "Give feedback" large panel
-  [ ] `vama_givefb_wv_close`: user clicks "Close" button in Webview opened in `vama_givefb_launch`
-  [ ] `vama_givefb_info`: User clicks "Learn more about the Veteran Usability Project" hyperlink in "Give feedback" large 

Based on Dylan's suggestions, I consolidated the original analytics events above to:

`vama_givefb_close ` and `vama_givefb_wv_close ` were combined into `vama_givefb_close` with the parameter `screenName`, and that should either be `Webview` or `InAppRecruitment`.

`vama_givefb_launch ` and `vama_givefb_info` were combined into `vama_givefb_open` with the parameter `linkType`, and that should either be `launch` or `info`.

## PR Checklist

  **Reviewer:** Confirm the items below as you review

- [ ] PR is connected to issue(s)
- [ ] Tests are included to cover this change (when possible)
- [ ] No magic strings (All string unions follow the [Union -> Constant](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/constants/common.ts) type pattern)
- [ ] No secrets or API keys are checked in
- [ ] All imports are absolute (no relative imports)
- [ ] New functions and Redux work have proper TSDoc annotations

## For QA

[Run a build for this branch](https://github.com/department-of-veterans-affairs/va-mobile-app/actions/workflows/on_demand_build.yml)

### CODE REVIEW

File: VAMobile/src/screens/HomeScreen/ProfileScreen/SettingsScreen/InAppRecruitmentScreen/InAppRecruitmentScreen.tsx [hunk 1]
```diff
@@ -16,9 +19,30 @@ const { LINK_URL_IN_APP_RECRUITMENT, LINK_URL_VETERAN_USABILITY_PROJECT } = getE
...
+    // Track when the user is leaving the Webview
+    if (isWebviewOpen && currentScreen.name !== 'Webview') {
+      logAnalyticsEvent(Events.vama_givefb_wv_close())
+      setIsWebviewOpen(false)
+    }
+  }, [isWebviewOpen, currentScreen])
 
   const onPress = () => {
-    navigation.navigate('Webview', { url: LINK_URL_IN_APP_RECRUITMENT, displayTitle: t('webview.vagov'), loadingMessage: t('inAppRecruitment.goToQuestionnaire.loading') })
+    logAnalyticsEvent(Events.vama_givefb_launch())
```
Comment (Sparowhawk) [inline VAMobile/src/screens/HomeScreen/ProfileScreen/SettingsScreen/InAppRecruitmentScreen/InAppRecruitmentScreen.tsx:39]: I wonder if we should change this analytic to a generic and apply it to more sections than just this with some parameters to differentiate the two

File: VAMobile/src/screens/HomeScreen/ProfileScreen/SettingsScreen/InAppRecruitmentScreen/InAppRecruitmentScreen.tsx [hunk 2]
```diff
@@ -53,6 +77,7 @@ const InAppRecruitmentScreen: FC<InAppRecruitmentScreenProps> = ({ navigation })
             numberOrUrlLink={LINK_URL_VETERAN_USABILITY_PROJECT}
             linkType={LinkTypeOptionsConstants.url}
             a11yLabel={t('inAppRecruitment.learnMore')}
+            fireAnalytic={() => logAnalyticsEvent(Events.vama_givefb_info())}
```
Comment (Sparowhawk) [inline VAMobile/src/screens/HomeScreen/ProfileScreen/SettingsScreen/InAppRecruitmentScreen/InAppRecruitmentScreen.tsx:80]: I wonder if we should change this analytic to a generic and apply it to more sections than just this with some parameters to differentiate the two

File: VAMobile/src/screens/HomeScreen/ProfileScreen/SettingsScreen/InAppRecruitmentScreen/InAppRecruitmentScreen.tsx [hunk 3]
```diff
@@ -16,9 +19,30 @@ const { LINK_URL_IN_APP_RECRUITMENT, LINK_URL_VETERAN_USABILITY_PROJECT } = getE
 const InAppRecruitmentScreen: FC<InAppRecruitmentScreenProps> = ({ navigation }) => {
   const { t } = useTranslation(NAMESPACE.COMMON)
   const theme = useTheme()
+  const navigateTo = useRouteNavigation()
+  const currentScreen = useNavigationState((state) => state.routes[state.routes.length - 1])
+  const [isWebviewOpen, setIsWebviewOpen] = useState(false)
+
+  useBeforeNavBackListener(navigation, () => {
+    logAnalyticsEvent(Events.vama_givefb_close())
```
Comment (Sparowhawk) [inline VAMobile/src/screens/HomeScreen/ProfileScreen/SettingsScreen/InAppRecruitmentScreen/InAppRecruitmentScreen.tsx:27]: I wonder if we should change this analytic to a generic and apply it to more sections than just this with some parameters to differentiate the two

File: VAMobile/src/screens/HomeScreen/ProfileScreen/SettingsScreen/InAppRecruitmentScreen/InAppRecruitmentScreen.tsx [hunk 4]
```diff
@@ -16,9 +19,30 @@ const { LINK_URL_IN_APP_RECRUITMENT, LINK_URL_VETERAN_USABILITY_PROJECT } = getE
...
+  })
+
+  useEffect(() => {
+    // Track when the user is leaving the Webview
+    if (isWebviewOpen && currentScreen.name !== 'Webview') {
+      logAnalyticsEvent(Events.vama_givefb_wv_close())
```
[the remark this answers is not in the document]
Comment (theodur) [reply VAMobile/src/screens/HomeScreen/ProfileScreen/SettingsScreen/InAppRecruitmentScreen/InAppRecruitmentScreen.tsx:33]: Do you mean combining `vama_givefb_close` and `vama_givefb_wv_close` into a single event, and passing the parameters like `Events.vama_givefb_close('webview')`, `Events.vama_givefb_close('screen')`?

[the remark this answers is not in the document]
Comment (Sparowhawk) [reply VAMobile/src/screens/HomeScreen/ProfileScreen/SettingsScreen/InAppRecruitmentScreen/InAppRecruitmentScreen.tsx:33]: yeah stuff like that would be preferred. If you can make it even more generic that would be great

[the remark this answers is not in the document]
Comment (theodur) [reply VAMobile/src/screens/HomeScreen/ProfileScreen/SettingsScreen/InAppRecruitmentScreen/InAppRecruitmentScreen.tsx:33]: I agree, that makes sense from a code perspective. I'll ask Greta to see if they're okay with that

[the remark this answers is not in the document]
Comment (theodur) [reply VAMobile/src/screens/HomeScreen/ProfileScreen/SettingsScreen/InAppRecruitmentScreen/InAppRecruitmentScreen.tsx]: Greta said it's okay to consolidate the analytics. Just pushed those changes up

**Review by Sparowhawk — APPROVED** [review]

**Review by DJUltraTom — APPROVED** [review]

**Pull request merged**